crates/pink/pink-extension/src/chain_extension.rs

```diff
@@ -104,105 +107,520 @@ pub type BatchHttpResult = Result<Vec<Result<HttpResponse, HttpRequestError>>, H
+    /// let value = pink::ext().cache_get(key);
+    /// ```
+    ///
+    /// # Availability
+    /// any contract | query
     #[ink(extension = 8, handle_status = false)]
     fn cache_get(key: &[u8]) -> Option<Vec<u8>>;
 
     /// Remove a value from the local cache.
     ///
-    /// Returns the removed value if it existed. Always returns `None` if it is called from a
-    /// command context.
+    /// This method removes a value from the local cache and returns the removed value if it existed.
+    /// If called from a command context, it will always return `None`.
+    ///
+    /// # Arguments
+    ///
+    /// * `args`: The key used to identify the value in the cache.
+    ///
+    /// # Returns
+    ///
+    /// * `Option<Vec<u8>>` - The removed value as a byte vector wrapped in an Option
+    ///     or `None` if the value did not exist or called in transaction.
+    ///
+    /// # Availability
+    /// any contract | query | transaction
     #[ink(extension = 9, handle_status = false)]
     fn cache_remove(args: &[u8]) -> Option<Vec<u8>>;
 
-    /// Print log message.
+    /// Log a message.
+    ///
+    /// This method logs a message at a given level.
+    ///
+    /// # Arguments
+    ///
+    /// * `level`: The level of the log message.
+    /// * `message`: The message to be logged.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let level = 1;
+    /// let message = "Hello, world!";
+    /// pink::ext().log(level, message);
+    /// ```
+    ///
+    /// # Note
+    /// This is the low-level method for logging. It is recommended to use shortcuts macros below instead:
+    ///
+    /// - [`debug!`]
+    /// - [`info!`]
+    /// - [`warn!`]
+    /// - [`error!`]
+    ///
+    /// # Availability
+    /// any contract | query | transaction
     #[ink(extension = 10, handle_status = false)]
     fn log(level: u8, message: &str);
 
-    /// Get random bytes, for query only
+    /// Get random bytes.
+    ///
+    /// This method generates a vector of random bytes of a given length. It returns random bytes
+    /// generated by hardware RNG. So it is not deterministic and only available in a query context.
+    ///
+    /// # Note
+    /// It always returns an empty vec![] when called in a transaction.
+    ///
+    ///
+    /// # Arguments
+    ///
+    /// * `length`: The length of the random bytes vector.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<u8>` - A vector of random bytes of the given length.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
+    /// let length = 32;
+    /// let random_bytes = pink::ext().getrandom(length);
+    /// ```
+    ///
+    /// # Availability
+    /// any contract | query only
     #[ink(extension = 11, handle_status = false)]
     fn getrandom(length: u8) -> Vec<u8>;
 
     /// Check if it is running in a Command context.
+    ///
+    /// # Returns
+    ///
+    /// * `bool` - `true` if it is running in a Command context, `false` if in query.
```

Comment:
Let's replace all the "command" context with "transaction" for easier understanding

Reply:
Searched and replaced some.